### PR TITLE
test: add luatest.server helpers for integrity module

### PR DESCRIPTION
**Pull request by Lord-KA**
This patch introduces new helpers for luatest.server to support `box.ctl.iproto_lockdown()` in tests. The lockdown drops all connections including the current one, so the helper reconnects with the server.

Part of #585

NO_DOC=test helper upgrade
NO_CHANGELOG=see NO_DOC
NO_TEST=see NO_DOC

### CODE REVIEW

**Review by ochaplashkin — APPROVED** [review]

File: test/luatest_helpers/server.lua [hunk 1]
```diff
@@ -190,4 +190,28 @@ function Server:start(opts)
...
+    if self.net_box == nil then
+        error('net_box is not connected', 2)
+    end
+    local ok, err = pcall(self.net_box.eval, self.net_box, code, ...)
+    if not ok and tostring(err):find('Peer closed') then
+        self:wait_until_ready()
+    elseif not ok then
+        error(err)
+    end
+    return err
```
Comment (Totktonada) [inline test/luatest_helpers/server.lua:205]: It looks a bit strange: the error is either raised or returned depending of whether the reconnect was occured.

File: test/luatest_helpers/server.lua [hunk 2]
```diff
@@ -190,4 +190,28 @@ function Server:start(opts)
     getmetatable(getmetatable(self)).start(self, opts)
 end
 
+-- Eval a string similar to `Server:eval()` and reconnect to the server
+-- in case the connection has been dropped.
+function Server:eval_reconnect(code, ...)
```
Comment (Totktonada) [inline test/luatest_helpers/server.lua:195]: Is it really a general purpose function?

A function for retrying a request with an on demand reconnection seems meaningful (but requires request_timeout, time_budget, retry_count and so on). But the reconnect without retrying looks quite specific.

Maybe it is better to include it into a test.

File: test/luatest_helpers/server.lua [hunk 3]
```diff
@@ -190,4 +190,28 @@ function Server:start(opts)
...
+end
+
+-- Prepare for iproto_lockdown caused by integrity check failure.
+function Server:prepare_integrity_lockdown()
+    -- Drop connection to the server ahead of time, before it is dropped
+    -- uncontrollingly in test-body. If the iproto_lockdown is already on,
+    -- it won't drop connections until turned off and on again.
+
+    self:eval_reconnect("box.ctl.iproto_lockdown(true)")
+end
```
Comment (Totktonada) [inline test/luatest_helpers/server.lua:215]: It is related to one specific feature. How about including it into a test or into a test helper inside the given test suite?

File: test/luatest_helpers/server.lua (hunk 1 above)
Comment (Lord-KA) [reply test/luatest_helpers/server.lua:205]: Right, renamed `err` to `res`.

**Comment by Lord-KA** [comment]
The helpers are moved to EE, so superseded by PR https://github.com/tarantool/tarantool-ee/pull/610.

**Pull request closed without merging**